hail/python/requirements.txt

```diff
@@ -1,35 +1,20 @@
-aiohttp>=3.8.1,<4
-aiohttp_session>=2.7,<2.8
+-r hailtop/requirements.txt
```

Comment:
@daniel-goldstein @danking This seems to break the `init_notebook.py` initialization action of `hailctl dataproc start`:
```
pip packages are ['setuptools', 'mkl<2020', 'lxml<5', 'google-cloud-storage==1.25.*', 'https://github.com/hail-is/jgscm/archive/v0.1.12+hail.zip', 'ipykernel==4.10.*', 'ipywidgets==7.5.*', 'jupyter-console==6.0.*', 'nbconvert==5.6.*', 'notebook==5.7.*', 'qtconsole==4.5.*', 'sklearn', 'slackclient', 'google', 'cpg-utils', '-r hailtop/requirements.txt', 'asyncinit>=0.2.4,<0.3', 'avro>=1.10,<1.12', 'bokeh==1.4.0', 'decorator<5', 'Deprecated>=1.2.10,<1.3', 'dill>=0.3.1.1,<0.4', 'frozenlist>=1.3.1,<2', 'hurry.filesize>=0.9,<1', 'Jinja2==3.0.3', 'nest_asyncio>=1.5.4,<2', 'numpy<2', 'pandas>=1.3.0,<1.5.0', 'parsimonious<0.9', 'plotly>=5.5.0,<5.11', 'protobuf==3.20.2', 'PyJWT', 'requests>=2.25.1,<3', 'scipy>1.2,<1.10', 'selenium', 'click', 'fsspec', 'gcloud', 'gnomad']
ERROR: Could not open requirements file: [Errno 2] No such file or directory: ' hailtop/requirements.txt'
```

Reply:
Our proposed quick fix :) https://github.com/populationgenomics/hail/pull/250

Reply:
Our proposed fix: https://github.com/populationgenomics/hail/pull/250/files

Reply:
Ah thanks for spotting this @lgruen and @vladsavelyev and sorry it broke the installation! I'll have to look into why the dataproc test passed.

Looks like you found the root cause, but I think a more proper fix would be to use `python/pinned-requirements.txt` instead of `python/requirements.txt`. The `pinned-requirements.txt` file should include the full set of requirements from hailtop and the rest of the hail package. Moreover, those will be fully pinned and tested versions that pass CI instead of some possibly new versions you get from a pip install. Does using that file instead of also including the hailtop requirements work?

Reply:
I've created a PR over at #12510 to try to get a fix in before an upcoming release.